src/wdb/QueryRunner.cc

```diff
@@ -364,6 +361,18 @@ bool QueryRunner::_dispatchChannel() {
             } else {
                 LOGS(_log, LOG_LVL_ERROR, "QR transmit taskSched == nullptr");
             }
+            // ATTENTION: This call is needed to record the _actual_ completion time of the task.
+            // It rewrites the finish timestamp within the task that was made when the task got
+            // kicked off the scheduler (see the code block above where a value of _removedFromThreadPool
+            // gets tested) which is happening shortly after MySQL query finishes and before the data
+            // transmission to Czar starts.
+            // NOTE: Tasks would stay in the task "cemetery" (class wpublish::QueriesAndChunks)
+            // for about 5 minutes after they finish transmitting data. After that no info on
+            // the task is available.
+            // TODO: Investigate an option for recording state transitions of the persistent
+            // metadata store of the worker, or keeping the state transisitons in a separate transient
+            // stopre that won't be affected by the task destruction.
```

Comment:
stopre -> store

How long to hold onto this information is worth some discussion. It's helpful, but there's a lot of it. Keeping statistics for query duration and such per table/chunk in a central location could be useful.